alert-common/src/main/java/com/synopsys/integration/alert/common/descriptor/config/ui/ProviderDistributionUIConfig.java

```diff
@@ -163,6 +163,17 @@ private ValidationResult validateConfigExists(FieldValueModel fieldToValidate, F
         }
     }
 
+    private ValidationResult validateFilterByProject(FieldValueModel fieldToValidate, FieldModel fieldModel) {
```

Comment:
This makes sure that if the User has checked `filterByProject` that they have to specify either a project name pattern or select a project.

Reply:
There is an odd difference between validation of new jobs and existing jobs. The UI will send empty fields to the back end and they get validated but editing a job does not do the same. If the project name pattern or configured projects have no values, they are not included in the payload for the validate request.